src/librustdoc/html/render/mod.rs

```diff
@@ -1048,11 +1048,15 @@ themePicker.onblur = handleThemeButtonsBlur;
     {
         let mut v = String::from("var searchIndex = JSON.parse('{\\\n");
         v.push_str(&all_indexes.join(",\\\n"));
-        // "addSearchOptions" has to be called first so the crate filtering can be set before the
-        // search might start (if it's set into the URL for example).
-        v.push_str("\\\n}');\naddSearchOptions(searchIndex);initSearch(searchIndex);");
+        v.push_str("\\\n}');\ninitSearch(searchIndex);");
         cx.shared.fs.write(&dst, &v)?;
     }
+
+    let crate_list_dst = cx.dst.join(&format!("crates{}.js", cx.shared.resource_suffix));
+    let crate_list =
+        format!("const crates = [{}];", krates.iter().map(|k| format!("\"{}\"", k)).join(","));
```

Comment:
`const` isn't part of ES5, please use instead: `window.CRATES = [{}]`. Also: this update is problematic because you remove crates that aren't listed, which is why you don't have `test` listed in the crates. Please update the list instead of rewriting it completely.

Reply:
In case you wonder: it's why we update the search index like that and have a crate define on each line separately.